include/pybind11/stl_bind.h

```diff
@@ -424,13 +424,18 @@ pybind11::class_<std::map<Key, T, Compare, Allocator>, holder_type> bind_map(pyb
 
     cl.def("__getitem__",
 		   [](Map &m, const KeyType &k) -> MappedType {
-			   if (m.count(k)) return m[k];
+			   if (m.count(k)) return m.at(k);
```

Comment:
This could potentially require two lookups--one for the one for the `count`, and one for the element access, and the `at()` is a bit wasteful (as it checks existence again).  The following would avoid both, while being const-usable:

```
auto it = m.find(k);
if (it != m.end()) return *m;
else throw pybind11::key_error();
```